gnovm/pkg/gnolang/op_expressions.go

```diff
@@ -194,8 +194,12 @@ func (m *Machine) doOpRef() {
 			nv.Value = rv2
 		}
 	}
+	elt := xv.TV.T
```

Comment:
Are the changes in this file necessary? The `TV.T` doesn't appear to be consumed later during the equality check. The tests that have been added still pass with these changes excluded.